wiki/osu!_File_Formats/Osu_(file_format)/en.md

```diff
@@ -176,90 +179,206 @@ Example of an inherited Timing Point:
+
+A perfect circle slider could be represented as a center point, a radius, and two angles for convenience. See this source code for the conversion algorithm: [CircularArcApproximator.cs](https://github.com/ppy/osu/blob/master/osu.Game/Rulesets/Objects/CircularArcApproximator.cs).
+
+A **Bézier** slider is made of one or many Bézier curves, sharing common ends. The degree of each curve is arbitrary. The first control points is defined with *x* and *y*, and the other ones by *curvePoints*.
+
+To identify the separation between two curves, the intersection point is repeated. Consider the sequence ABCDDEFFG. You would get the 3 Bézier curves: ABCD (cubic), DEF (quadratic) , FG (linear).
 
-`424,96,66,2,0,B|380:120|332:96|332:96|304:124,1,130,2|0,0:0|0:0,0:0:0:0:`
+Note that this representation is ambiguous, because ABBBC could actually represent ABBBC, AB+BBC, AB+B+BC, or AB+BBC.
 
-*x*, *y*, *time*, and *type* behave the same as described in Hit Circle Syntax.
+Example: `476,340,6419,2,0,B|437:336|422:309|422:309|384:309|359:337|359:337|328:308|300:304|300:304|272:352|237:383|176:356|159:287|224:256,1,420,6|2,0:0|0:0,0:0:0:0:`
 
-*hitSound* applies to the body of the slider, but only whistle sounds will play during the slider body.
+The first Bézier curve in the segment is quadratic with the following points: (476, 340), (437, 336), (422, 309).
 
-*sliderType* will be `L` (linear), `P` (perfect), `B` (Bezier), or `C` (Catmull). A slider created in the editor with only a start and end point will be a linear slider. A slider with only its start, end, and one grey point will be a perfect circle slider. All others will be Bezier. Catmull sliders are deprecated.
+Note that the multiple curves in the same Bézier slider need not have the same length, so you *will* need to approximate their lengths, otherwise the slider ball would speed up or slow down brutally in the middle of a slider.
 
-*curvePoints (x:y|...)* is a series of `|`-separated coordinates describing the control points of the slider. Red points appear twice. NOTE: curvePoints is separated from sliderType with a `|`, not a comma.
+#### Repeat
 
-*repeat (Integer)* is the number of times a player will go over the slider. A value of 1 will not repeat, 2 will repeat once, 3 twice, and so on.
+*repeat* (Integer) is the number of times a player will go over the slider. A value of 1 will not repeat, 2 will repeat once, 3 twice, and so on.
+
+#### Duration
 
 *pixelLength (Float)* is the length of the slider along the path of the described curve. If the length is greater than that of the described curve, the slider will continue in a straight line.
 
-*edgeHitsounds (hitSound|...)* is a `|`-separated list of hitSounds to apply to the circles of the slider. The values are the same as those for Hit Circle hitSounds.
+TODO: What's the unit? Is it really pixels?
+
+TODO: What happens if a Bézier or perfect circle slider's *pixelLength* is too big? Would you still get a straight line at the end?
```

Comment:
perfect circle will extend the curve. bezier makes a straight line

Reply:
By the way, is it safe to assume the editor will always write the correct length?

Reply:
Aw, this is complicated.

What if the pixelLength is shorter than the curve? Is the curve shrinked?

To summarize, the parser shoud 1. parse the path, 2. compute its actual length, 3. extend the path until its length matches pixelLength. Right?

Also, why would anyone make the actual length and pixelLength differ? For a perfect circle, it's just about moving the end point, and for a Bézier curve it's about adding an explicit straight curve. I guess the answer is *historical reasons* here.

Reply:
for your parser, that sounds about right. but keep in mind that most sliders will be *shorter* than their underlying curve (when the length is shorter, it follows the same shaped path, but only as much as it can)

I think you're missing the point of this feature. without it, it'd be nearly impossible to create on-time sliders--- the editor is designed so that the mapper can build a curved path for a slider to sit on. the slider won't always fill the entire curve (in fact, it almost never will), but it'll fill as much as it needs to to snap in time with the currently selected beat divisor

so, there are three reasons that the curve length and pixelLength differ:
- the curve length is almost never the correct length to snap in time with the music. and that's the most important part of sliders
- placing points so that they *exactly* match to both the curve and the music is impossible for most sliders because it'd require the user to place points on a non-integral position
- if they didn't differ, why would it even be written here? you can calculate the length of any curve given just it's points

just try it in the editor yourself if you don't understand how it works. it's like this because it's most intuitive for the mapper, not because of historical reasons

Reply:
Got it! Thank you for the detailed explanation.